content/blaze.md

```diff
@@ -0,0 +1,543 @@
+```
+
+<h3 id="builtin-block-helpers">Builtin Block Helpers</h3>
+There are a few builtin block helpers that are worth knowing about:
+
+<h4 id="if-unless">If / Unless</h4>
+
+The `{{#if}}` and `{{#unless}}` helpers are fairly straightforward but invaluable for controlling the content generated by a template. Both operate by evaluating and checking their single argument for "falsey"-ness (in JS this means `null`, `undefined`, `0`, `''`, `[]` and of course `false`).
+
+```html
+{{#if something}}
+  <p>It's true</p>
+{{else}}
+  <p>It's false</p>
+{{/if}}
+```
+
+<h4 id="each-in">Each-in</h4>
+
+The `{{#each in}}` helper is a convenient way to step over a list whilst retaining the outer data context. 
+
+```html
+{{#each todo in todos}}
+{{/each}}
+```
+
+In this case `todo` will be added to the data context within the block, but all the existing data context items (`list` and `todosReady` in this case) will remain available.
+
+<h4 id="let">Let</h4>
+The `{{#let}}` helper is useful to capture the output of a helper or document subproperty within a template:
+
+```html
+{{#let name=person.bio.firstName color=generateColor}}
+  <div>{{name}} gets a {{color}} card!</div>
+{{/let}}
+```
+
+XXX: what to do about d.c. changing helpers such as with/each?
+
+<h4 id="strictness">Strictness</h4>
+Spacebars starts from a very strict interpretation of HTML. In particular you need to careful about self-closing tags and when you are allowed to use them. For instance, you can't self-close a `div` (`<div/>`) in Spacebars.
+
+<h4 id="escaping">Escaping</h4>
+To insert a literal {{, {{{, or any number of curly braces, put a vertical bar after it. So `{{|` will show up as `{{`, `{{{|` will show up as `{{{`, and so on.
+
+<h2 id="reusable-components">Creating reusable components in Blaze</h2>
+In <a href="ui-ux">the UI/UX article</a> we discussed the merits of creating reusable components that are modular and depend and interact with their environment in clear and minimal ways. 
+
+Although Blaze, as a simple template-based runtime, doesn't enforce a lot of these principles (as some other rendering frameworks, such as React, do) you can take steps to write your Blaze components in such a way that you can enjoy most of the same benefits. This section will attempt to outline some "best practice" for writing such Blaze components.
+
+Examples below will reference the `listShow` component from the Todos example app.
+
+<h3 id="validate-data-context">Validate data contexts</h3>
+In order to ensure your component is only used in the way you expect, you should validate the data context provided to it. The data context provides the inputs to the component and so should be checked.
+
+XXX: this does not yet work
+
+You can do this in the components `onCreated()` callback, like so:
+
+```js
+Template.listsShow.onCreated(function() {
+  this.autorun(() => {
+    check(Template.data(), new SimpleSchema({
+     list: {blackbox: true},
+     todosReady: {type: Boolean}
+   }));
```

Comment:
Yeah. We just need to make sure it actually works (right now it throws due to the `list` failing the object test)